scripts/lib/CIME/test_status.py

```diff
@@ -92,13 +92,23 @@ def _test_helper1(file_contents):
     ts._parse_test_status(file_contents) # pylint: disable=protected-access
     return ts._phase_statuses # pylint: disable=protected-access
 
-def _test_helper2(file_contents, wait_for_run=False, check_throughput=False, check_memory=False, ignore_namelists=False):
-    ts = TestStatus(test_dir="/", test_name="ERS.foo.A")
-    ts._parse_test_status(file_contents) # pylint: disable=protected-access
-    return ts.get_overall_test_status(wait_for_run=wait_for_run,
-                                      check_throughput=check_throughput,
-                                      check_memory=check_memory,
-                                      ignore_namelists=ignore_namelists)
+def _test_helper2(file_contents, wait_for_run=False, check_throughput=False, check_memory=False, ignore_namelists=False, no_run=False):
+    lines = file_contents.splitlines()
+    rv = None
+    for perm in itertools.permutations(lines):
+        ts = TestStatus(test_dir="/", test_name="ERS.foo.A")
+        ts._parse_test_status("\n".join(perm)) # pylint: disable=protected-access
```

Comment:
Big increase in robustness of testing. All permutations of phase orders are tested, they should all produce a consistent result.